factory/stateComponents.go

```diff
@@ -67,14 +73,56 @@ func (scf *stateComponentsFactory) Create() (*StateComponents, error) {
 		return nil, fmt.Errorf("%w for ValidatorPubkeyConverter: %s", ErrPubKeyConverterCreation, err.Error())
 	}
 
+	shardID := core.GetShardIDString(scf.shardCoordinator.SelfId())
```

Comment:
can you extract the accountsAdapter creation into a separate function and call it here?

Reply:
Done.